internal/controllers/gateway/route_utils.go

```diff
@@ -231,12 +243,14 @@ func getSupportedGatewayForRoute[T gatewayapi.RouteT](ctx context.Context, mgrc
 				if *parentRef.Port != listener.Port {
 					// This ParentRef has a port specified and it's different
 					// than current listener's port.
+					listenerLogger.V(util.DebugLevel).Info("Listener name does not match parentRef.Port")
```

Comment:
This might be rather verbose but let's leave it for now since it's debug. I suggest adding some field to the log message, otherwise it would be hard to figure out which listener and which parent ref are we talking about here.

```suggestion
					listenerLogger.V(util.DebugLevel).Info(
						"Listener port does not match parentRef.Port",
						"listener_port", listener.Port,	"parentRef_port", parentRef.Port,
						)
```